Lib/pathlib.py

```diff
@@ -54,13 +54,30 @@ def _ignore_error(exception):
             getattr(exception, 'winerror', None) in _IGNORED_WINERRORS)
 
 
+@functools.lru_cache()
```

Comment:
Drive-by nit: since we know there's only two flavours, you can use an unbounded cache here. It should be a teeny tiny bit faster (since the cache mechanism doesn't have to worry about popping the least-recently-used cache entries out of the cache), and it's slightly prettier to look at as well :)

```suggestion
@functools.cache
```